src/api/operations.ts

```diff
@@ -225,33 +225,85 @@ export const drawRectangle = (options: {
   graphicsState?: string | PDFName;
   matrix?: TransformationMatrix;
   clipSpaces?: Space[];
-}) => [
-    pushGraphicsState(),
-    options.graphicsState && setGraphicsState(options.graphicsState),
-    options.color && setFillingColor(options.color),
-    options.borderColor && setStrokingColor(options.borderColor),
-    setLineWidth(options.borderWidth),
-    options.borderLineCap && setLineCap(options.borderLineCap),
-    setDashPattern(options.borderDashArray ?? [], options.borderDashPhase ?? 0),
-    ...(options.clipSpaces ? clipSpaces(options.clipSpaces) : []),
-    options.matrix && concatTransformationMatrix(...options.matrix),
-    translate(options.x, options.y),
-    rotateRadians(toRadians(options.rotate)),
-    skewRadians(toRadians(options.xSkew), toRadians(options.ySkew)),
-    moveTo(0, 0),
-    lineTo(0, options.height),
-    lineTo(options.width, options.height),
-    lineTo(options.width, 0),
-    closePath(),
-
-    // prettier-ignore
-    options.color && options.borderWidth ? fillAndStroke()
+  radius?: number | PDFNumber;
+}) => {
+  if (!options.radius || asNumber(options.radius) <= 0) {
+    return [
+      pushGraphicsState(),
+      options.graphicsState && setGraphicsState(options.graphicsState),
+      options.color && setFillingColor(options.color),
+      options.borderColor && setStrokingColor(options.borderColor),
+      setLineWidth(options.borderWidth),
+      options.borderLineCap && setLineCap(options.borderLineCap),
+      setDashPattern(options.borderDashArray ?? [], options.borderDashPhase ?? 0),
+      ...(options.clipSpaces ? clipSpaces(options.clipSpaces) : []),
+      options.matrix && concatTransformationMatrix(...options.matrix),
+      translate(options.x, options.y),
+      rotateRadians(toRadians(options.rotate)),
+      skewRadians(toRadians(options.xSkew), toRadians(options.ySkew)),
+      moveTo(0, 0),
+      lineTo(0, options.height),
+      lineTo(options.width, options.height),
+      lineTo(options.width, 0),
+      closePath(),
+
+      // prettier-ignore
+      options.color && options.borderWidth ? fillAndStroke()
   : options.color                      ? fill()
   : options.borderColor                ? stroke()
-  : closePath(),
+            : closePath(),
+
+      popGraphicsState(),
+    ].filter(Boolean) as PDFOperator[];
+  } else {
+    const radius = asNumber(options.radius);
+    const width = asNumber(options.width);
+    const height = asNumber(options.height);
+
+    if (radius > width / 2.0 || radius > height / 2.0) {
+      console.warn("radius exceeds the rectangle's width or height");
```

Comment:
If the radius exceeds half of the rectangle's width/height, a warning is issued, and an empty array is returned. This causes the rendering to fail, but the error may not be clear.

Please adjust it to the maximum possible radius.